docs/developing/troubleshooting.md

```diff
@@ -63,3 +63,27 @@ If your app wants to write to locations other than `/var`, and it's not easy to
 change the app's code, one work-around is to create a symlink from the location
 you app wants to modify to a location under `/var`.  This won't work for all
 applications however.
+
+## ETag data
+
+Sandstorm drops invalid ETag data. This typically has no impact on app functionality. When Sandstorm
+drops invalid ETag data, it stores a warning in the grain log that it did so, limiting the log
+messages to at most one per launch of a grain to avoid filling up the log file. See also [issue
+#2295.](https://github.com/sandstorm-io/sandstorm/issues/2295)
+
+In versions v0.176 and earlier, the grain would crash, rather than log an error message, if an
```

Comment:
I'll also replace

> The headers are dropped within the grain at the

with

> Headers are processed within the grain at the

since I think that explains the paragraph better. I imagine you don't care strongly about that rewording.